### PR TITLE
Jetpack App: Updates the prologue screen

### LINKED ISSUE #16362: Jetpack App (Basics): Prologue Screen
prologue-examples-1-1

### Description
Update prologue screen design for the Jetpack app

### PR DESCRIPTION
**Pull request by emilylaguna**
WPAuthenticator Related PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/592

Fixes #16362

### Video
https://user-images.githubusercontent.com/793774/116590322-9b3a4400-a8eb-11eb-8400-2a4a0baf5b24.mov

### Screenshots

| iPhone | iPad |
|:---:|:---:|
|Screen Shot 2021-04-29 at 1 03 38 PM|Screen Shot 2021-04-29 at 1 03 39 PM|

### To test:
1. Build the app using the Jetpack scheme
2. Launch the app
3. If you're logged in, log out
4. You'll see the new prologue screen
5. Tap around the view
6. Verify the view looks the same in dark mode and light mode
7. Follow the testing steps in the related PR

### Regression Notes
**1. Potential unintended areas of impact**
This includes change to WPAuthenticator and the NUXButton which impact other areas of the app and WooCommerce

**2. What I did to test those areas of impact (or what existing automated tests I relied on)**

- Manually tested on WP scheme
- Manually tested WooCommerce iOS
- Tested the other areas of the app that the NUXButton is used

**3. What automated tests I added (or what prevented me from doing so)**
None, this is a UI change

### PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        let view = StarFieldView(with: config)
+        view.layer.masksToBounds = true
+        return view
+    }()
+
+    var gradientLayer: CALayer = {
+        let gradientLayer = CAGradientLayer()
+
+        // Start color is the background color with no alpha because if we use clear it will fade to black
+        // instead of just disappearin
```
Comment (leandroalonso) [inline WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift]:
```suggestion
        // instead of just disappearing
```

**Review by leandroalonso — APPROVED** [review]
@emilylaguna this looks gorgeous!

Tested in iPhone and iPad, also double-checked the NUXButton when logged in.

**Review by osullivanchris — APPROVED** [review]
Testing on an iPhone and reviewing iPad from screenshots. 

I love it! Amazing job.

File: WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        vertical.maximumRelativeValue = amount
+
+        let group = UIMotionEffectGroup()
+        group.motionEffects = [horizontal, vertical]
+
+        stackView.addMotionEffect(group)
+    }
+
+    private struct Constants {
+        static let parallaxAmount: CGFloat = 20
```
Comment (frosty) [inline WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift]: I love that this is parallaxed! :) I think we could perhaps push this up a little more? 30?

File: WordPress/Classes/Utility/App Configuration/AuthenticationHandler.swift [hunk 3]
```diff
@@ -4,4 +4,12 @@ protocol AuthenticationHandler {
     func shouldPresentUsernamePasswordController(for siteInfo: WordPressComSiteInfo?, onCompletion: @escaping (WordPressAuthenticatorResult) -> Void)
 
     func presentLoginEpilogue(in navigationController: UINavigationController, for credentials: AuthenticatorCredentials, onDismiss: @escaping () -> Void) -> Bool
+
+    // WPAuthenticator style overrides
+    var statusBarStyle: UIStatusBarStyle { get }
```
Comment (frosty) [inline WordPress/Classes/Utility/App Configuration/AuthenticationHandler.swift:9]: I'm not sure these 100% feel like they belong in here, but I don't have much of a better suggestion 🤷‍♂️

File: WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        horizontal.maximumRelativeValue = amount
+
+        let vertical = UIInterpolatingMotionEffect(keyPath: "center.y", type: .tiltAlongVerticalAxis)
+        vertical.minimumRelativeValue = -amount
+        vertical.maximumRelativeValue = amount
+
+        let group = UIMotionEffectGroup()
+        group.motionEffects = [horizontal, vertical]
+
+        stackView.addMotionEffect(group)
```
Comment (frosty) [inline WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift:85]: I also tested out adding this to the starfield, but I noticed you can see the particles falling off the edge of the view when it's displaced.

**Review by frosty — APPROVED** [review]
This looks wonderful! 🤩 I wonder if there's anything we can do to improve handling of large dynamic type in the center:



That's an exaggerated example, but the title gets pushed off. I wonder if we can do something with constraints to at least keep Jetpack on the screen.

File: WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift (hunk 2 above)
Comment (emilylaguna) [reply WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift]: More parallax!? Sounds good!

File: WordPress/Classes/Utility/App Configuration/AuthenticationHandler.swift (hunk 3 above)
Comment (emilylaguna) [reply WordPress/Classes/Utility/App Configuration/AuthenticationHandler.swift:9]: I agree, and I didn't have one either. I can always move them in the future.

File: WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift (hunk 2 above)
Comment (emilylaguna) [reply WordPress/Jetpack/Classes/NUX/JetpackPrologueViewController.swift]: Done in: de6f25e0d88ed5d5c50b1b74bea13733f6c22368

**Comment by emilylaguna** [comment]
@frosty To adjust for accessibility I: 
- Added a max height of the label
- If the font size is too large I hide the label. I'd rather hide it than show truncated giant text. 

Done in: c5129d7f9e5faa6c25b97054b3aceef0426d1d3d
